drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+        )
+    )
+
+    # extract all memory dumps temporarily into a dumps/ folder
+    with zipfile.ZipFile(analysis_dir / "dumps.zip", "r") as zip_ref:
+        try:
+            # try to extract the memory dumps into the analysis folder
+            zip_ref.extractall(analysis_dir)
+            dumps = analysis_dir / "dumps"
+        except PermissionError:
+            # in case of missing permissions, extract into /tmp
+            zip_ref.extractall(Path("/tmp"))
+            dumps = Path("/tmp") / "dumps"
+
+    # extract the capabilities within each memory dump, one per thread
+    pool = multiprocessing.Pool(processes=len(malware_dumps))
+    yield from pool.starmap(
+        static_capa_analysis, map(lambda dump: (dumps / dump, rules), malware_dumps)
+    )
+
+    # try to remove the temporarily created folder
+    try:
+        shutil.rmtree(dumps)
+    except PermissionError:
+        logger.debug(
+            "Permission Denied: Could not remove temporary dumps.zip extraction folder"
+        )
+
+
+def format_capa_address(address: Union[Tuple, capa.features.address.Address]) -> Dict:
+    # this method formats capa address (in the format of tuples) into a single dictionary
+    if isinstance(address, tuple):
+        return functools.reduce(
+            lambda a, b: a | b, [format_capa_address(addr) for addr in address]
+        )
+    elif isinstance(
+        address,
+        (
+            capa.features.address.AbsoluteVirtualAddress,
+            capa.features.address.RelativeVirtualAddress,
+            capa.features.address.FileOffsetAddress,
+        ),
+    ):
+        return {"address": hex(address)}
+    elif isinstance(address, capa.features.address.DNTokenAddress):
+        return {"token": address.token}
+    elif isinstance(address, capa.features.address.DNTokenOffsetAddress):
+        return {**format_capa_address(address.token), "offset": address.offset}
+    elif isinstance(address, capa.features.address.ProcessAddress):
+        return {"ppid": address.ppid, "pid": address.pid}
+    elif isinstance(address, capa.features.address.ThreadAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.process)}
+    elif isinstance(address, capa.features.address.DynamicCallAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.thread)}
+    elif isinstance(address, capa.features.address._NoAddress):
+        # empty address
+        return {}
+    else:
+        logger.debug("Encountered unknown address type: %s", type(address))
+        return {"address": address}
+
+
+def construct_ttp_block(
+    rule: capa.rules.Rule, addresses: capa.features.address
+) -> Dict[str, Any]:
+    name = rule.name.split("/")[0]
+    mbc = rule.meta.get("mbc", None)
+    attck = rule.meta.get("att&ck", None)
+    occurrences = [format_capa_address(address=address) for address, _ in addresses]
```

Comment:
You use addresses as tuple, but the type in method signature is capa.features.address. I think this is just a bug in typing, so type in the method signature should be update.

Reply:
Yeah it's an issue with typing, I've fixed it now. It's `List[Tuple[ca.Address, Result]]`